src/settings/settings_manager.cpp

```diff
@@ -112,10 +113,25 @@ void SettingsManager::DefineSetting(SettingId id, const std::string &name,
                                     const type::Value &value,
                                     const std::string &description,
                                     const type::Value &default_value,
+                                    const type::Value &min_value,
+                                    const type::Value &max_value,
                                     bool is_mutable, bool is_persistent) {
   if (settings_.find(id) != settings_.end()) {
     throw SettingsException("settings " + name + " already exists");
   }
+
+  // Only below types support min-max bound checking
+  if (value.GetTypeId() == type::TypeId::INTEGER ||
+      value.GetTypeId() == type::TypeId::SMALLINT ||
+      value.GetTypeId() == type::TypeId::TINYINT ||
+      value.GetTypeId() == type::TypeId::DECIMAL) {
+    if (!value.CompareBetweenInclusive(min_value, max_value))
+      throw SettingsException("Value given for \"" + name +
+                              "\" is not in its min-max bounds (" +
+                              std::to_string(min_value.GetAs<int32_t>()) + "-" +
```

Comment:
Is this part correct? 
The range checking looks ok, but if it fails, the min and max values are converted to int32_t in order to report the error. Can't decimal type can be larger than int32_t?

Reply:
Aha! Yup, this is incorrect indeed. Just updated in latest patch.